cns/service/main.go

```diff
@@ -805,6 +828,25 @@ func main() {
 		}
 	}
 
+	_, envEnableAsyncPodDelete := os.LookupEnv(envVarEnableAsyncPodDelete)
+	if cnsconfig.EnableAsyncPodDelete || envEnableAsyncPodDelete {
+		// Start fs watcher here
+		logger.Printf("[Azure CNS] Start fsnotify watcher for intended deletes")
+		cnsclient, err := cnsclient.New(cnsBaseURL, cnsReqTimeout) //nolint
+		if err != nil {
+			logger.Errorf("failed to initialize CNS client:%v.\n", err)
+		}
+		w := &fsnotify.Watcher{
+			CnsClient: cnsclient,
+		}
+		watcherPath := cnsconfig.WatcherPath
+		watcherDirectory := cnsconfig.DeleteDirectory
+		err = fsnotify.WatchFs(w, watcherPath, watcherDirectory, z)
+		if err != nil {
+			logger.Errorf("Failed to start fsnotify watcher, err:%v.\n", err)
```

Comment:
same here, should we exit if this fails? i think we should be retrying, at least